core/base/src/main/java/alluxio/util/io/PathUtils.java

```diff
@@ -171,6 +173,58 @@ public static String getParent(String path) throws InvalidPathException {
     return parent;
   }
 
+  /**
+   * Join two path elements for ufs, separated by {@link AlluxioURI#SEPARATOR}.
+   *
+   * For example,
+   *
+   * <pre>
+   * {@code
+   * concatUfsPath("s3://myroot/", "filename").equals("s3://myroot/filename");
+   * concatUfsPath("s3://", "filename").equals("s3://filename");
+   * }
+   * </pre>
+   *
+   * @param base base path
+   * @param path path element to concatenate
+   * @return joined path
+   */
+  public static String concatUfsPath(String base, String path) {
+    Pattern basePattern = Pattern.compile("(...)\\/\\/");
+    // use conCatPath to join path if base is not starting with //
+    if (!basePattern.matcher(base).matches()) {
+      return concatPath(base, path);
+    } else {
+      Preconditions.checkArgument(base != null, "Failed to concatPath: base is null");
+      Preconditions.checkArgument(path != null, "Failed to concatPath: a null set of paths");
+      String trimmedPath = SEPARATOR_MATCHER.trimFrom(path);
+      StringBuilder output = new StringBuilder(base.length() + trimmedPath.length());
+      output.append(base);
+      output.append(trimmedPath);
+      return output.toString();
+    }
+  }
+
+  /**
+   * Get temp path for async persistence job.
+   *
+   * @param path ufs path
+   * @return ufs temp path with UUID
+   */
+  public static String getPersistentTmpPath(String path) {
+    StringBuilder tempFilePath = new StringBuilder();
+    StringBuilder tempFileName = new StringBuilder();
+    String fileName = FilenameUtils.getName(path);
+    tempFilePath.append(".alluxio_ufs_persistence/");
+    String uuid = UUID.randomUUID().toString();
+    tempFileName.append(fileName);
```

Comment:
would it be useful to have the date in the prefix of the filenames? Then, it could be easy to see which files are "old"?